src/ast/query.rs

```diff
@@ -916,7 +924,7 @@ pub enum TableFactor {
         /// This field's value is `Some(v)`, where `v` is a (possibly empty)
         /// vector of arguments, in the case of a table-valued function call,
         /// whereas it's `None` in the case of a regular table name.
-        args: Option<Vec<FunctionArg>>,
+        args: Option<TableFunctionArgs>,
```

Comment:
instead of introducing a new type, can we add a new variant to [FunctionArg](https://github.com/sqlparser-rs/sqlparser-rs/blob/c48f157995d5c57a970aa1c91f3045f5b653d3b1/src/ast/mod.rs#L4739-L4746) and update the parser to conditionally accept it for [clickhouse/generic](https://github.com/sqlparser-rs/sqlparser-rs/blob/c48f157995d5c57a970aa1c91f3045f5b653d3b1/src/parser/mod.rs#L10290)?
I think that would let us reuse existing types and be a less invasive change.

e.g.
```rust
pub enum FunctionArg {
    Named {
        name: Ident,
        arg: FunctionArgExpr,
        operator: FunctionArgOperator,
    },
    Unnamed(FunctionArgExpr),
    Settings {
        name: Ident, arg: Expr
    }
}
```

Reply:
The reason I went with a new type is twofold:
 - The `SETTINGS` clause is a single clause, at the end of the args list
 - It is only allowed on table-valued functions

Another option is to add a `settings: Vec<Setting>` field to `TableFactor::Table`. That loses the information that it's part of the argument list, but retains the single-clause and table-valued function only aspect and is more compatible.
wdyt?

Reply:
Ah I see, yeah the current representation sounds reasonable given settings is a single clause